compiler/modules/front.syntax/ast.bal

```diff
@@ -542,7 +542,7 @@ public type SingletonTypeDesc record {|
     (string|float|int|boolean|decimal) value;
 |};
 
-public type SubsetBuiltinTypeName "any"|"boolean"|"int"|"float"|"string"|"error";
+public type SubsetBuiltinTypeName "any"|"boolean"|"int"|"float"|"decimal"|"string"|"error";
```

Comment:
You need to remove `decimal` from BuiltinTypeName

Reply:
Removed